apis_core/relations/templatetags/relations.py

```diff
@@ -58,11 +58,16 @@ def relations_from(from_obj, relation_type: ContentType = None):
 
 @register.simple_tag(takes_context=True)
 def relations_list_table(context, relations, target=None):
-    table_modules = module_paths(
-        type(context["object"]),
-        path="tables",
-        suffix=f"{target.name.capitalize()}RelationsTable",
-    )
+    suffixes = ["RelationsTable"]
+    if target:
+        suffixes.extend(
+            f"{module[-1]}RelationsTable" for module in mro_paths(target.model_class())
+        )
+    table_modules = ()
+    for suffix in suffixes:
+        table_modules += module_paths(
+            type(context["object"]), path="tables", suffix=suffix
+        )
```

Comment:
could we put that \"find the table class\" functionality in a dedicated function and add a cache?